dot/digest/digest_test.go

```diff
@@ -515,3 +516,73 @@ func TestHandler_HandleNextConfigData(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, act.ToConfigData(), stored)
 }
+
+func Test_persistBABEDigestsForNextEpoch(t *testing.T) {
+	type testStruct struct {
+		defineNextEpochData          bool
+		defineNextConfigData         bool
+		expectedErr                  error
+		alreadyDefinedFuncThrowError bool
```

Comment:
nit you could define your mocks using functional fields if you want. [example](https://gist.github.com/qdm12/99358e2529a5887ae13a936a7f1c8ae5#file-potato_test-go-L230)

Reply:
Nice suggestion, shrink the test conditions